src/workerd/jsg/modules.h

```diff
@@ -52,6 +52,108 @@ class CommonJsModuleContext: public jsg::Object {
   jsg::Value exports;
 };
 
+class NodeJsModuleContext: public jsg::Object {
+  // The NodeJsModuleContext is similar in structure to CommonJsModuleContext
+  // with the exception that:
+  // (a) Node.js-compat built-in modules can be required without the `node:` specifier-prefix
+  //     (meaning that worker-bundle modules whose names conflict with the Node.js built-ins
+  //     are ignored), and
+  // (b) The common Node.js globals that we implement are exposed. For instance, `process`
+  //     and `Buffer` will be found at the global scope.
+public:
+  NodeJsModuleContext(v8::Isolate* isolate, kj::Path path)
+      : module(jsg::alloc<CommonJsModuleObject>(isolate)), path(kj::mv(path)),
+      exports(isolate, module->getExports(isolate)) {}
+
+  v8::Local<v8::Value> require(kj::String specifier, v8::Isolate* isolate);
+  v8::Local<v8::Value> getBuffer(jsg::Lock& js);
+  v8::Local<v8::Value> getProcess(jsg::Lock& js);
+
+  jsg::Ref<CommonJsModuleObject> getModule(v8::Isolate* isolate) { return module.addRef(); }
+
+  v8::Local<v8::Value> getExports(v8::Isolate* isolate) { return exports.getHandle(isolate); }
+  void setExports(jsg::Value value) { exports = kj::mv(value); }
+
+  JSG_RESOURCE_TYPE(NodeJsModuleContext) {
+    JSG_METHOD(require);
+    JSG_READONLY_INSTANCE_PROPERTY(module, getModule);
+    JSG_INSTANCE_PROPERTY(exports, getExports, setExports);
+    JSG_LAZY_INSTANCE_PROPERTY(Buffer, getBuffer);
+    JSG_LAZY_INSTANCE_PROPERTY(process, getProcess);
+  }
+
+  // The additional Node.js globals include...
```

Comment:
This is one of the more complex bits. We likely just need this context to extend from `ServiceWorkerGlobalScope` so we can pick these additional bits up...

OR

Should we just bite the bullet and say that if the the `nodejs_compat` flag is enabled, the additional Node.js globals (e.g. `process`, `Buffer`, `require()` without specifier-prefix etc )are Just Always Available as part of `ServiceWorkerGlobalScope`? Maybe we use a separate compat flag to enable those?

Thoughts?